refactor: Require request access for all communities

0x-r4bbit opened the pull request:
This commit introduces a few changes regarding users accessing
communities:

1. While the APIs still exist, community invites should no longer be
   used, instead communities should merely be "shared".
2. Sharing a community to users allows users to "join" the community,
   which in reality makes them request access to that community.
3. This means, users have to request access to any community, even if
   the community has permissions set to `NO_MEMBERSHIP`
4. Only difference between `ON_REQUEST` and `NO_MEMBERSHIP` is that
   `ON_REQUEST` communities require manual approval of the owner/admin
   to access a community. `NO_MEMBERSHIP` communities accept
   automatically (as soon as owner/admin receives the request).

This also implies that users are no longer optimistically added to the
member list of communities, but only after they have been accepted.

This introduces a bit of a message ping-pong for users to know that
someone is now part of a community

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status_go.html))?
- [x] Have you tested changes with mobile?
- [x] Have you tested changes with desktop?

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (51)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | feeada51 | [#1](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/1/) | 2022-05-20 10:17:06 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220520-101446-feeada-pr2682.zip) |
| :heavy_check_mark: | feeada51 | [#1](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/1/) | 2022-05-20 10:19:12 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220520-101446-feeada-pr2682.zip) |
| :heavy_check_mark: | feeada51 | [#1](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/1/) | 2022-05-20 10:19:37 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220520-101446-feeada-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 1cb3ac84 | [#2](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/2/) | 2022-05-20 10:25:33 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220520-102338-1cb3ac-pr2682.zip) |
| :heavy_check_mark: | 1cb3ac84 | [#2](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/2/) | 2022-05-20 10:25:54 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220520-102341-1cb3ac-pr2682.zip) |
| :heavy_check_mark: | 1cb3ac84 | [#2](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/2/) | 2022-05-20 10:26:52 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220520-102338-1cb3ac-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | e37f63fc | [#3](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/3/) | 2022-05-20 11:02:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220520-110129-e37f63-pr2682.zip) |
| :heavy_check_mark: | e37f63fc | [#3](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/3/) | 2022-05-20 11:03:30 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220520-110129-e37f63-pr2682.zip) |
| :heavy_check_mark: | e37f63fc | [#3](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/3/) | 2022-05-20 11:04:31 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220520-110126-e37f63-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 1f007d3b | [#4](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/4/) | 2022-05-20 12:19:43 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220520-121817-1f007d-pr2682.zip) |
| :heavy_check_mark: | 1f007d3b | [#4](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/4/) | 2022-05-20 12:21:20 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220520-121816-1f007d-pr2682.aar) |
| :heavy_check_mark: | 1f007d3b | [#4](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/4/) | 2022-05-20 12:22:02 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220520-121817-1f007d-pr2682.zip) |
| | | | | | | |
| :heavy_check_mark: | efd0fc57 | [#5](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/5/) | 2022-05-23 08:59:16 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220523-085748-efd0fc-pr2682.zip) |
| :heavy_check_mark: | efd0fc57 | [#5](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/5/) | 2022-05-23 09:00:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220523-085744-efd0fc-pr2682.zip) |
| :heavy_check_mark: | efd0fc57 | [#5](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/5/) | 2022-05-23 09:02:32 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220523-085744-efd0fc-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | fad1759b | [#6](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/6/) | 2022-05-23 14:01:55 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220523-135940-fad175-pr2682.zip) |
| :heavy_check_mark: | fad1759b | [#6](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/6/) | 2022-05-23 14:04:27 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220523-135940-fad175-pr2682.zip) |
| :heavy_check_mark: | fad1759b | [#6](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/6/) | 2022-05-23 14:05:15 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220523-135940-fad175-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 5ce96cd1 | [#7](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/7/) | 2022-05-24 07:18:01 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220524-071547-5ce96c-pr2682.zip) |
| :heavy_check_mark: | 5ce96cd1 | [#7](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/7/) | 2022-05-24 07:24:55 | ~9 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220524-071550-5ce96c-pr2682.zip) |
| :heavy_check_mark: | 5ce96cd1 | [#7](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/7/) | 2022-05-24 07:25:05 | ~9 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220524-071547-5ce96c-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 586ad835 | [#8](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/8/) | 2022-05-24 11:17:42 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220524-111518-586ad8-pr2682.zip) |
| :heavy_check_mark: | 586ad835 | [#8](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/8/) | 2022-05-24 11:22:08 | ~6 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220524-111515-586ad8-pr2682.aar) |
| :heavy_check_mark: | 586ad835 | [#8](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/8/) | 2022-05-24 11:25:34 | ~10 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220524-111518-586ad8-pr2682.zip) |
| | | | | | | |
| :heavy_check_mark: | e4082350 | [#9](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/9/) | 2022-05-24 11:50:51 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220524-114906-e40823-pr2682.zip) |
| :heavy_check_mark: | e4082350 | [#9](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/9/) | 2022-05-24 11:56:18 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220524-114905-e40823-pr2682.aar) |
| :heavy_check_mark: | e4082350 | [#9](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/9/) | 2022-05-24 12:09:30 | ~20 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220524-114909-e40823-pr2682.zip) |
| | | | | | | |
| :x: | 5e2edf54 | [#10](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/10/) | 2022-05-30 14:16:51 | ~53 sec | `linux` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/10/consoleText) |
| :x: | 5e2edf54 | [#10](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/10/) | 2022-05-30 14:17:07 | ~1 min | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/10/consoleText) |
| :x: | 5e2edf54 | [#10](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/10/) | 2022-05-30 14:17:29 | ~1 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/10/consoleText) |
| | | | | | | |
| :heavy_check_mark: | 1ff60872 | [#11](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/11/) | 2022-05-31 10:08:32 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-100621-1ff608-pr2682.zip) |
| :heavy_check_mark: | 1ff60872 | [#11](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/11/) | 2022-05-31 10:08:41 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-100621-1ff608-pr2682.zip) |
| :heavy_check_mark: | 1ff60872 | [#11](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/11/) | 2022-05-31 10:09:45 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-100619-1ff608-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | d43c5a34 | [#12](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/12/) | 2022-05-31 10:42:59 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-104050-d43c5a-pr2682.zip) |
| :heavy_check_mark: | d43c5a34 | [#12](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/12/) | 2022-05-31 10:43:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-104050-d43c5a-pr2682.zip) |
| :heavy_check_mark: | d43c5a34 | [#12](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/12/) | 2022-05-31 10:48:06 | ~7 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-104050-d43c5a-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 59974102 | [#13](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/13/) | 2022-05-31 10:45:49 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220531-104316-599741-pr2682.zip) |
| :heavy_check_mark: | 59974102 | [#13](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/13/) | 2022-05-31 10:47:21 | ~4 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220531-104303-599741-pr2682.zip) |
| :heavy_check_mark: | 59974102 | [#13](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/13/) | 2022-05-31 10:52:12 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220531-104809-599741-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 7294bc97 | [#14](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/14/) | 2022-06-01 17:07:48 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220601-170559-7294bc-pr2682.zip) |
| :heavy_check_mark: | 7294bc97 | [#14](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/14/) | 2022-06-01 17:09:09 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220601-170559-7294bc-pr2682.zip) |
| :heavy_check_mark: | 7294bc97 | [#14](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/14/) | 2022-06-01 17:09:30 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220601-170558-7294bc-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | c7b83d2d | [#15](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/15/) | 2022-06-16 11:11:26 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220616-110925-c7b83d-pr2682.zip) |
| :heavy_check_mark: | c7b83d2d | [#15](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/15/) | 2022-06-16 11:12:48 | ~3 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220616-110922-c7b83d-pr2682.zip) |
| :heavy_check_mark: | c7b83d2d | [#15](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/15/) | 2022-06-16 11:14:44 | ~5 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220616-110922-c7b83d-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 690daf3a | [#16](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/16/) | 2022-06-17 13:01:12 | ~2 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220617-125859-690daf-pr2682.zip) |
| :heavy_check_mark: | 690daf3a | [#16](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/16/) | 2022-06-17 13:01:24 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220617-125901-690daf-pr2682.zip) |
| :heavy_check_mark: | 690daf3a | [#16](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/16/) | 2022-06-17 13:02:35 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220617-125859-690daf-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | bc6f1e1c | [#17](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/17/) | 2022-06-17 19:16:33 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220617-191458-bc6f1e-pr2682.zip) |
| :heavy_check_mark: | bc6f1e1c | [#17](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/17/) | 2022-06-17 19:19:42 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220617-191455-bc6f1e-pr2682.aar) |
| :heavy_check_mark: | bc6f1e1c | [#17](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/17/) | 2022-06-17 19:20:16 | ~5 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220617-191458-bc6f1e-pr2682.zip) |
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | d3fcf4d3 | [#18](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/18/) | 2022-07-04 09:44:00 | ~2 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220704-094141-d3fcf4-pr2682.zip) |
| :heavy_check_mark: | d3fcf4d3 | [#18](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/18/) | 2022-07-04 09:46:03 | ~4 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220704-094140-d3fcf4-pr2682.zip) |
| :heavy_check_mark: | d3fcf4d3 | [#18](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/18/) | 2022-07-04 09:46:35 | ~4 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220704-094140-d3fcf4-pr2682.aar) |
| | | | | | | |
| :heavy_check_mark: | 88439fd9 | [#19](https://ci.status.im/job/status-go/job/prs/job/linux/job/PR-2682/19/) | 2022-07-08 10:59:35 | ~1 min | `linux` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-desktop-220708-105801-88439f-pr2682.zip) |
| :heavy_check_mark: | 88439fd9 | [#19](https://ci.status.im/job/status-go/job/prs/job/ios/job/PR-2682/19/) | 2022-07-08 10:59:56 | ~1 min | `ios` | [:package:`zip`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-ios-220708-105801-88439f-pr2682.zip) |
| :heavy_check_mark: | 88439fd9 | [#19](https://ci.status.im/job/status-go/job/prs/job/android/job/PR-2682/19/) | 2022-07-08 11:01:23 | ~3 min | `android` | [:package:`aar`](https://status-im-prs.ams3.cdn.digitaloceanspaces.com/status-go-android-220708-105801-88439f-pr2682.aar) |

0x-r4bbit commented on protocol/communities_messenger_test.go:
```diff
@@ -618,15 +620,13 @@ func (s *MessengerCommunitiesSuite) TestImportCommunity() {
 		}
+		if !response.Communities()[0].IsAdmin() {
+			return errors.New("isn't admin despite import")
+		}
```
I tried putting the `Joined()` check for alice in here as well, but it always makes the test fail in a weird way.

cammellos replied:
looking into it, would you mind pushing the failed test if you have time please?

0x-r4bbit replied:
It's already pushed.

The tests that are failing are `TestRequestAccess` and `TestRequestAccessAgain`

Let me know if I can be of any assistance and thanks for taking the time!

0x-r4bbit reviewed (commented):
@cammellos @Samyoul sorry to bug you with this, but I just can't seem to figure out how to make the tests pass in this one.

The issue still seems to be that the accept-to-join-responses are sent/processed asynchronously, and therefore are not part of the `response` that's being tested.

I tried putting in `time.Sleep()` calls to have some time pass but that either makes the tests time out or just fail with the same error.

This is working when testing it in actual desktop instances, just having trouble getting this simulated properly in the testsuite :(

0x-r4bbit commented on protocol/communities/manager.go:
```diff
@@ -819,9 +840,63 @@ func (m *Manager) HandleCommunityRequestToJoin(signer *ecdsa.PublicKey, request
+	metadataMessage := &protobuf.ApplicationMetadataMessage{
+		Payload: communityDescriptionBytes,
+		Type:    protobuf.ApplicationMetadataMessage_COMMUNITY_DESCRIPTION,
+	}
```
@Samyoul Any objections here to leave the signature out?

Samyoul replied:
This is fine

0x-r4bbit commented:
BTW I'll squash the second commit into the first one once this has been reviewed, so that we don't introduce a commit that breaks .
I've split it up for now so it's easier to review, so no need to review the second commit.

saledjenic commented on protocol/communities_messenger_test.go:
```diff
@@ -574,6 +574,8 @@ func (s *MessengerCommunitiesSuite) TestImportCommunity() {
 	s.Require().NotNil(response)
 	s.Require().Len(response.Communities(), 1)
 	s.Require().Len(response.CommunitiesSettings(), 1)
+	s.Require().True(response.Communities()[0].Joined())
```
@PascalPrecht isn't a bit dangerous accessing here with direct index without checking before that `if len(response.Communities()) > 0`?

0x-r4bbit replied:
There's 

> 	s.Require().Len(response.Communities(), 1)

Two lines above that (L575), which should do the trick, no?

saledjenic replied:
@PascalPrecht yes, you're right, missed that somehow even I looked around this line.

Samyoul commented on protocol/protobuf/communities.proto:
```diff
@@ -83,9 +83,10 @@ message CommunityRequestToJoin {
-  CommunityDescription community = 2;
-  bool accepted = 3;
-  bytes grant = 4;
+  bytes community_id = 2;
+  CommunityDescription community = 3;
+  bool accepted = 4;
+  bytes grant = 5;
```
This is a breaking change. If an older client sends this message type it will be parsed incorrectly by a newer client, if an older client receives this message from a newer client it will be parsed incorrectly by the older client.

0x-r4bbit replied:
Good catch!
This is not a problem if we introduce `community_id` as field `5` and leave the rest as is, right?

Samyoul replied:
Yes correct

0x-r4bbit replied:
Updated

Samyoul reviewed (approved):
I've spent a bit of time going through this and this looks great.

0x-r4bbit commented:
@qoqobolo @churik This one is ready for testing on mobile!

elina2015 commented:
@richard-ramos could you kindly build  https://github.com/status-im/status-desktop/pull/6082  for Desktop testing of this ticket?

0x-r4bbit commented:
The failing tests here seem unrelated to the changes (the PR has passed test stage before, I've only rebased it due to conflicts in generated bindata)

0x-r4bbit committed:
refactor: Require request access for all communities

This commit introduces a few changes regarding users accessing
communities:

While the APIs still exist, community invites should no longer be
used, instead communities should merely be "shared".
Sharing a community to users allows users to "join" the community,
which in reality makes them request access to that community.
This means, users have to request access to any community, even if
the community has permissions set to NO_MEMBERSHIP
Only difference between ON_REQUEST and NO_MEMBERSHIP is that
ON_REQUEST communities require manual approval of the owner/admin
to access a community. NO_MEMBERSHIP communities accept
automatically (as soon as owner/admin receives the request).
This also implies that users are no longer optimistically added to the
member list of communities, but only after they have been accepted.

This introduces a bit of a message ping-pong for users to know that
someone is now part of a community
